packages/frontend/src/hooks/useTransaction.ts

```diff
@@ -0,0 +1,238 @@
+            const l1BridgeInterface = new Interface(l1BridgeAbi)
+            const decodedData: Result = l1BridgeInterface?.decodeFunctionData(
+              methodName,
+              response.data
+            )
+
+            if (decodedData) {
+              params = formatLogArgs(decodedData)
+
+              const { amount, deadline, chainId, recipient } = decodedData
+              destNetworkName = networkIdToSlug(chainId)
+
+              const transferFromL1Completeds = await fetchTransferFromL1Completeds(
+                destNetworkName,
+                recipient,
+                amount,
+                deadline
+              )
+
+              if (transferFromL1Completeds?.length) {
+                const lastTransfer: L1Transfer = getLastLog(transferFromL1Completeds)
+                const { token, transactionHash, timestamp } = lastTransfer
+                ts = timestamp
+                tokenSymbol = token
+                destTxHash = transactionHash
+                destExplorerLink = getExplorerTxUrl(destNetworkName, transactionHash)
+                eventValues = lastTransfer
+              }
+            }
+          }
+
+          if (methodName === 'swapAndSend') {
+            txType = TxType.sendL2ToL2
+            const ammWrapperInterface = new Interface(l2AmmWrapperAbi)
+            const decodedData = ammWrapperInterface.decodeFunctionData(methodName, response.data)
+            if (decodedData) {
+              params = formatLogArgs(decodedData)
+              destNetworkName = networkIdToSlug(params.chainId)
+
+              const tsDetails = getTransferSentDetailsFromLogs(receipt.logs)
+              const { chainId, transferId } = tsDetails!
+              destNetworkName = networkIdToSlug(chainId)
+
+              const withdrawalBondeds = await fetchWithdrawalBondedsByTransferId(
+                destNetworkName,
+                transferId!
+              )
+              if (withdrawalBondeds?.length) {
+                const wbDetails = getLastLog(withdrawalBondeds)
+                console.log(`wbDetails:`, wbDetails)
+                const { transactionHash, token, timestamp } = wbDetails
+                ts = timestamp
+                tokenSymbol = token
+                destTxHash = transactionHash
+                destExplorerLink = getExplorerTxUrl(destNetworkName, transactionHash)
+                eventValues = wbDetails
+              }
+            }
+          }
+
+          const waitConfirmations = getNetworkWaitConfirmations(slug as string)
+          setNetworkConfirmations(waitConfirmations)
+          setConfirmations(response.confirmations)
+
+          if (waitConfirmations && response.confirmations >= waitConfirmations) {
+            setCompleted(true)
+          }
+
+          setTxObj({
+            type: txType,
+            response,
+            receipt,
+            methodName,
+            params,
+            gasCost,
+            datetime: new Date(ts * 1000).toString(),
```

Comment:
are we assuming that `ts` is always defined here?

Reply:
will change to `new Date(parseInt(timestamp || '0') * 1000).toString()`